posts/2023-10-24-broken-badges-and-23k-keywords.md

```diff
@@ -0,0 +1,40 @@
+---
+layout: post
+title: A tale of broken badges and 23,000 features
+author: Tobias Bieniek
+team: the crates.io team <https://www.rust-lang.org/governance/teams/crates-io>
+---
+
+Around mid-October of 2023 we, the crates.io team, were [notified](https://github.com/rust-lang/crates.io/issues/7269) by one of our users that a [shields.io](https://shields.io) badge for their crate stopped working. The issue reporter was kind enough to already debug the problem and figured out that the API request that shields.io sends to crates.io was most likely the problem. Here is a quote from the original issue:
+
+> This crate makes heavy use of feature flags which bloats the response payload of the API.
+
+Apparently the API response for this specific crate had broken the 20 MB mark and shields.io wasn't particularly happy with this. Interestingly, this crate only had 9 versions published at this point in time. But how do you get to 20 MB with only 9 published versions?
+
+As the quote above already mentions, this crate is using features… a lot of features… almost 23,000 features to be precise! 😱
+
+What crate needs that many features? Well, this crate provides SVG icons for Rust-based web applications… and it uses one feature per icon so that the payload size of the final WebAssembly bundle stays small.
+
+At first glance there should be nothing wrong with this. This seems like a reasonable thing to do from a crate author perspective and neither cargo, nor crates.io were showing any warnings about this. Unfortunately, some of the implementation details are not too happy about such high numbers of features though…
```

Comment:
Similar "trailing text" with "though"

And one of these two are correct style, and which one you think is correct depends on which style guide you are:
```suggestion
At first glance there should be nothing wrong with this. This seems like a reasonable thing to do from a crate author perspective and neither cargo nor crates.io were showing any warnings about this. Unfortunately, some of the implementation details are not too happy about such high numbers of features…
```

```suggestion
At first glance there should be nothing wrong with this. This seems like a reasonable thing to do from a crate author perspective and neither cargo, nor crates.io, were showing any warnings about this. Unfortunately, some of the implementation details are not too happy about such high numbers of features…
```